src/js/angular/clustermanagement/controllers.js

```diff
@@ -548,3 +642,73 @@ function AddLocationCtrl($scope, $modalInstance, toastr, productInfo, $translate
 const getDocBase = function (productInfo) {
     return `https://graphdb.ontotext.com/documentation/${productInfo.productShortVersion}/${productInfo.productType}`;
 };
+
+AddNodesDialogCtrl.$inject = ['$scope', '$modalInstance', 'data', '$modal', 'RemoteLocationsService'];
+
+function AddNodesDialogCtrl($scope, $modalInstance, data, $modal, RemoteLocationsService) {
+    const clusterConfiguration = angular.copy(data.clusterConfiguration);
+    const clusterModel = angular.copy(data.clusterModel);
+    $scope.nodes = [];
+
+    $scope.clusterNodes = clusterModel.locations.filter((location) => clusterConfiguration.nodes.includes(location.rpcAddress));
+    $scope.locations = clusterModel.locations.filter((location) => !$scope.clusterNodes.includes(location));
+    $scope.locations.forEach((location) => location.isNew = true);
```

Comment:
Can't you chain this with the filter above?

Reply:
No, as the consumer does not return the array.